x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/serializer/serializer.ts

```diff
@@ -65,6 +65,8 @@ export const createSerializer = (originalPolicy?: SerializedPolicy) => (
           delete hotPhaseActions.forcemerge;
         } else if (_meta.hot.bestCompression) {
           hotPhaseActions.forcemerge!.index_codec = 'best_compression';
+        } else {
+          delete hotPhaseActions.forcemerge!.index_codec;
         }
 
         if (_meta.hot.bestCompression && hotPhaseActions.forcemerge) {
```

Comment:
I wasn't sure how to test this logic in the UI and if there's any additional logic required here too.

Reply:
Yeah, it is a bit more tricky to fully test in the UI since we have to interact with the component and submit the form to catch the network request that would have been made and then make assertions against that payload. Kind of like this https://github.com/elastic/kibana/blob/b99ca969e02dc0f67835cee4dfcabc505e7c7395/x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/edit_policy.test.ts#L119

In this case I guess we'd make a policy that has this enabled, submit a request, check that it is there, then disable and send the request again. Although that might require wrestling with any routing that happens too.

For now, I think having a unit test for the serialisation is good, since we know the toggle can actually set the value. The trickier (more bug prone) part of the serialiser is removing values that have been unset.